homeassistant/components/miraie_ac/climate.py

```diff
@@ -0,0 +1,303 @@
+
+import logging
+from typing import Any
+
+from py_miraie_ac import (
+    Device as MirAIeDevice,
+    MirAIeAPI,
+    PowerMode as MirAIePowerMode,
+    SwingMode as MirAIeSwingMode,
+)
+
+from homeassistant.components.climate import (
+    FAN_OFF,
+    PRECISION_WHOLE,
+    PRESET_NONE,
+    SWING_BOTH,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+    SWING_VERTICAL,
+    ClimateEntity,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN
+from .const import (
+    FAN_MODE_MAP_TO_HASS,
+    FAN_MODE_MAP_TO_MIRAIE,
+    HVAC_MODE_MAP_TO_HASS,
+    HVAC_MODE_MAP_TO_MIRAIE,
+    MAX_TEMP,
+    MIN_TEMP,
+    PRESET_MODE_MAP_TO_HASS,
+    PRESET_MODE_MAP_TO_MIRAIE,
+    SUPPORTED_FAN_MODES,
+    SUPPORTED_FEATURES,
+    SUPPORTED_HVAC_MODES,
+    SUPPORTED_PRESET_MODES,
+    SUPPORTED_SWING_MODES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add MirAIe AC devices."""
+    try:
+        api: MirAIeAPI = hass.data[DOMAIN][entry.entry_id]
+        await api.initialize()
+        devices: list[MirAIeDevice] = api.devices
+
+        for device in devices:
+            _LOGGER.debug("Found MirAIe device: %s", device.friendly_name)
+
+        entities = list(map(MirAIeClimateEntity, devices))
+        async_add_entities(entities)
+    except Exception as exp:
+        _LOGGER.error("Unexpected error in MirAIe AC: %s", exp)
+        raise HomeAssistantError(f"Unexpected error in MirAIe AC: {exp}") from Exception
+
+
+class MirAIeClimateEntity(ClimateEntity):
+    """Define MirAIe Climate."""
+
+    def __init__(self, device: MirAIeDevice) -> None:
+        """Initialize MirAIe climate entity."""
+        self._attr_hvac_modes = SUPPORTED_HVAC_MODES
+        self._attr_preset_modes = SUPPORTED_PRESET_MODES
+        self._attr_fan_modes = SUPPORTED_FAN_MODES
+        self._attr_swing_modes = SUPPORTED_SWING_MODES
+        self._attr_supported_features = SUPPORTED_FEATURES
+        self._attr_fan_mode = FAN_OFF
+        self._attr_max_temp = MAX_TEMP
+        self._attr_min_temp = MIN_TEMP
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_precision = PRECISION_WHOLE
+        self._attr_unique_id = device.device_id
+        self._friendly_name = device.friendly_name
+        self.device = device
+
+        _LOGGER.debug("MirAIe device added: %s", device.friendly_name)
+
+    @property
+    def should_poll(self) -> bool:
+        """Let Hass know that polling is not required."""
+        return False
+
+    @property
+    def name(self) -> str:
```

Comment:
New integrations are required to implement the [new naming format](https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations).